### PR TITLE
Remove unused payment API actions

### PR DESCRIPTION
**Pull request by mbj**
Extraction from #398

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

**Comment by mbj** [comment]
@dkubb I confirmed via our main apps logs that these actions where never hit, also the fact no backend changes had to be done confirms this.

File: api/config/routes.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
     resources :orders, only: [] do
       resources :line_items, only: %i[destroy update]
-      resources :payments do
+      resources :payments, only: %i[show update] do
```
Comment (dkubb) [inline api/config/routes.rb]: This change is not strictly related to the commit it appears in, which is more about removal. I would suggest breaking this out into it's own commit.

Comment (mbj) [inline api/config/routes.rb]: Actually its very closely related. We do not want the routes to be generated for the actions we removed.

We remove `create` and `index`, and without this `only` option they'd be routable.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.
